docs/operate/control/headless-app.md

```diff
@@ -5,5 +5,532 @@ weight: 30
+- You want to reduce latency, for example for running [PID control loops](https://en.wikipedia.org/wiki/Proportional%E2%80%93integral%E2%80%93derivative_controller), or
+- Your machine runs continuously (for example, an air quality sensor) and it is impractical to constantly run a client from your development computer
+
+## Install an SDK
+
+Install your preferred Viam SDK on the Linux or macOS computer SBC where you plan to run the control script.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+If you are using the Python SDK, [set up a virtual environment](/sdks/python/python-venv/) to package the SDK inside before running your code, avoiding conflicts with other projects or your system.
+
+For macOS (both Intel `x86_64` and Apple Silicon) or Linux (`x86`, `aarch64`, `armv6l`), run the following commands:
+
+```sh {class="command-line" data-prompt="$"}
+python3 -m venv .venv
+source .venv/bin/activate
+pip install viam-sdk
+```
+
+Windows is not supported.
+If you are using Windows, use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and install the Python SDK using the preceding instructions for Linux.
+For other unsupported systems, see [Installing from source](https://python.viam.dev/#installing-from-source).
+
+If you intend to use the [ML (machine learning) model service](/services/ml/), use the following command instead, which installs additional required dependencies along with the Python SDK:
+
+```sh {class="command-line" data-prompt="$"}
+pip install 'viam-sdk[mlmodel]'
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+Run the following command to install the [Viam Go SDK](https://pkg.go.dev/go.viam.com/rdk):
+
+```sh {class="command-line" data-prompt="$"}
+go get go.viam.com/rdk/robot/client
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+Follow the [instructions on the GitHub repository](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md).
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Authenticate your script
+
+To connect your code to your machine, use the API keys Viam automatically generates for you.
+
+1. On your machine's page in the [Viam app](https://app.viam.com), click the **CONNECT** tab.
+
+1. Select your preferred programming **Language**.
+   A fully functional code snippet will be displayed, with connection code as well as some calls to the APIs of the resources you've configured on your machine.
+   The machine address (which resembles `12345.somemachine-main.viam.cloud`) auto-populates into the connection code.
+   It is a public address to connect to your machine.
+
+1. Toggle the **Include API key** to auto-populate the API key and API key ID into the code.
+   You can either:
+
+   - Copy the code with the API key in it, **OR**
+   - To reduce the risk of accidentally sharing your API key when collaborating on code, create environment variables for the API key and API key ID, and copy the code with **Include API key** toggled off.
```

Comment:
I think this is going in the direction of explaining UI possibly to much. We may be able to simplify. I think it's also worth considering where we point people to API keys that have broader scope. And we probably need to create content for API keys in reference or somewhere else. 
```suggestion
1. Copy the sample code.
   We strongly recommend to store your API keys in environment variables to reduce (...).
   If your code will connect to multiple machines or use [Platform APIs](/dev/reference/apis/#platform-ap) you can [create an API key with broader access](TODO).  
```

Reply:
That is fair. And then whatever we end up with on this page probably gets added in the mobile and web app pages too. I think API key content makes the most sense in Control Access, and then we'd link to that from here, essentially saying "you can use the machine API key to access a single machine, or for access to multiple machines, read about location-level and org-level API keys here"